Description: syslinux 6 update
 Patch is reworked, based on the work of Daniel Baumann.
 syslinux 6.x has a few incompatible changes:
  * upstream: with the addition of efi support, syslinux modules are
    now stored in a different location, e.g:
      /usr/lib/syslinux/vesamenu.c32 ->
      /usr/lib/syslinux/modules/bios/vesamenu.c32
 .
  * upstream: since syslinux 5.x, some bootloaders need additional
    loader files (ldlinux.*) to be copied to the medium.
    for syslinux/extlinux this is already handled automatically,
    for the 'manual' bootloaders (isolinux/pxelinux), this needs to
    be done manually.
 .
  * upstream: since syslinux 5.x, modules have dependencies, e.g:
      using vesamenu.c32, you also need libutil.c32 and libcom32.c32
    for more information, see:
    http://www.syslinux.org/wiki/index.php/Library_modules
 .
  * debian: with the addition of efi support, syslinux bootloaders
    are packaged separately, e.g.:
    isolinux.bin is in isolinux, pxelinux is in pxelinux, etc.
    packages re-using the syslinux bootloaders should depend on
    these individual bootloader packages.
 .
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Bug-Debian: http://bugs.debian.org/746751
Last-Update: 2014-05-03

---

--- unetbootin-603.orig/unetbootin.cpp
+++ unetbootin-603/unetbootin.cpp
@@ -3376,13 +3376,21 @@ void unetbootin::instIndvfl(QString srcf
 	if (srcfName == "memdisk")
 				srcF.setFileName(QFile::exists("/usr/share/syslinux/memdisk") ? "/usr/share/syslinux/memdisk" : "/usr/lib/syslinux/memdisk");
 	else if (srcfName == "menu.c32")
+	{
 				srcF.setFileName(QFile::exists("/usr/share/syslinux/menu.c32") ? "/usr/share/syslinux/menu.c32" : "/usr/lib/syslinux/menu.c32");
+				if (QFile::exists("/usr/lib/syslinux/bios/menu.c32"))
+					srcF.setFileName("/usr/lib/syslinux/bios/menu.c32");
+	}
     else if (srcfName == "libutil.c32")
                 srcF.setFileName(QFile::exists("/usr/share/syslinux/libutil.c32") ? "/usr/share/syslinux/libutil.c32" : "/usr/lib/syslinux/libutil.c32");
     else if (srcfName == "libcom32.c32")
                 srcF.setFileName(QFile::exists("/usr/share/syslinux/libcom32.c32") ? "/usr/share/syslinux/libcom32.c32" : "/usr/lib/syslinux/libcom32.c32");
 	else if (srcfName == "mbr.bin")
+	{
 				srcF.setFileName(QFile::exists("/usr/share/syslinux/mbr.bin") ? "/usr/share/syslinux/mbr.bin" : "/usr/lib/syslinux/mbr.bin");
+				if (QFile::exists("/usr/lib/syslinux/mbr/mbr.bin"))
+					srcF.setFileName("/usr/lib/syslinux/mbr/mbr.bin");
+	}
 	else if (srcfName == "ubnsylnx")
 		srcF.setFileName("/usr/bin/syslinux");
 //	else
@@ -4118,6 +4126,8 @@ void unetbootin::runinstusb()
 			QFile mbrbinF(":/mbr.bin");
 			#ifdef NOSTATIC
 			mbrbinF.setFileName(QFile::exists("/usr/share/syslinux/mbr.bin") ? "/usr/share/syslinux/mbr.bin" : "/usr/lib/syslinux/mbr.bin");
+			if (QFile::exists("/usr/lib/syslinux/mbr/mbr.bin"))
+				mbrbinF.setFileName("/usr/lib/syslinux/mbr/mbr.bin");
 			#endif
 			usbmbrF.open(QIODevice::WriteOnly);
 			mbrbinF.open(QIODevice::ReadOnly);
